### ISSUE TITLE
Persona not working for a user on geeksphone keon running FirefoxOS 1.0.1

### ISSUE THREAD

**Issue by lloyd** [mozilla/persona#3500]
@dougt reports that signing into persona sites (specifically ones pointed at firefoxos.persona.org) is not working.  that he's getting stuck inside the persona dialog at the 'signing in' screen.

he's tried http://whereismyfox.com and marketplace and both demonstrate this problem.

dougt is on a production geeksphone keon running 1.0.1

He also reports console messages (which @zaach believes are orthogonal):

```
E/GeckoConsole(  812): [JavaScript Error: "redeclaration of const kIdentityDelegateWatch" {file: "chrome://browser/content/identity.js" line: 36}]
```

**Comment by lloyd** [comment]
5 star treatment so we can figure out what is going on here.

**Comment by lloyd** [comment]
123done.org works for dougt in that he can get through the dialog (the assertion verification fails because we're still on firefoxos.persona.org for the moment)

so it looks like this problem is specific to hosted apps on a 1.0.1 device.  Does anyone have any context here?

**Comment by kumar303** [comment]
I believe Keon's ship with an earlier build (that would suck). We are running the latest 1.0.1 build for development and no issues here.

**Comment by lloyd** [comment]
can a keon user update to the latest 1.0.1?  how can dougt check his uh micro version thingy?

**Comment by jedp** [comment]
That `redeclaration` error is ignorable.  (I think it's due to a separate gecko bug in which the initial page load event is fired twice)

**Comment by kumar303** [comment]
I think Settings > Device > More Info will have git commit info but I'm not sure. It does for my 1.0.1 build. Example: last commit was 2013-06-07 10:47:43 on the 1.0.1 branch.

**Comment by lloyd** [comment]
@dougt - please provide the above version information

**Comment by edwindotcom** [comment]
nice script to get version info: https://wiki.mozilla.org/File:CheckVersions.zip

**Comment by chilts** [comment]
I have a 1.0.1.0-prerelease phone. I'm connected to wifi here in the office.

# What I did
1. Opened the browser
2. opened whereismyfox.com
3. Clicked "Sign in with Persona"
4. Entered my personal email address (which is already in Person) and clicked "Continue"
5. The next screen then says "Create a Password" and has two inputs - further down it says "Your email address is new to us." etc

I would have expected it to be a screen saying "Enter your password" since my email address is known.

Is this consistent with what @dougt or @zaach sees.

(I haven't got it connected to my computer at the moment, but just thought I'd give it a try and relay my observations.)

**Comment by lloyd** [comment]
@chilts - this is a non-issue.  currently firefoxos uses a different server than our production deployment, we'll move to a single deployment early next week.

**Comment by chilts** [comment]
Ok. np. :)

**Comment by lloyd** [comment]
@jedp offers:

The Keon's are going to be hard to debug.  We don't know whether they have the latest persona fixes or not on their device.  They could even be hitting native-persona.org.  Ugh.

Who has a keon?

Good Keon user, can you push this pref to it:   pref("toolkit.identity.debug", true);

Go to firefoxos.123done.org, and send a complete "adb logcat"?

**Comment by dougt** [comment]
Logs here:
    http://pastebin.mozilla.org/2496452

**Comment by lloyd** [comment]
@vladikoff reports that nightly 1.0.1 builds are working without a problem - version nightly-images-keon-2013-06-05.Gecko-00ce01b.Gaia-0c0f4c2

**Comment by kumar303** [comment]
if those are release build hashes (I think they are) then we have:

https://git.mozilla.org/?p=releases/gecko.git;a=commitdiff;h=00ce01b
https://git.mozilla.org/?p=releases/gaia.git;a=commitdiff;h=0c0f4c2

...both from June 4th. That build should have a working persona. Maybe there is something interesting in logcat with `pref("toolkit.identity.debug", true);` as Jed says.

**Comment by lloyd** [comment]
**recap** - specific to keon, specific to the build that they shipped, specific to hosted apps

**we need** 
1. (@dougt) logs while logging into _marketplace_
2. (@vladikoff) a report whether either of the builds posted on http://downloads.geeksphone.com/ fix the issue
3. deeper understanding of the issue, so we can see if we can mitigate it somehow on our end.

**Comment by dougt** [comment]
This is the log while logging into marketplace: http://pastebin.mozilla.org/2496561

**Comment by jedp** [comment]
@dougt I see on line 213 that it's sent the `watch` command, but I don't see the logs for marketplace's login `request` there - is there any more to that logcat?  (You should be able to grep for `Identity` to trim things down)

**Comment by dougt** [comment]
That is everything.

**Comment by chilts** [comment]
Ok, so I updated the Keon via an alert on the phone itself. ie. System update (with an optional maps update too).

Version is now :
- Software/OS version : Boot2Gecko 1.0.1.0-prerelease
- Last Updated: 06/07/2013 3:13pm
- Platform version: 18.0
- Build Identifier: 20130603111001
- Update channel: nightly
- Git commit info: 2013-06-03 06:39:48 - fcae2365

Once installed and rebooted, I managed to log in to Marketplace fine. I was already logged into Persona so just clicked "Sign In", rather than having to enter my password.

Any other info anyone needs? I have a logcat too if anyone wants it.

**Comment by ozten** [comment]
Is line 237 common and ignorable, or the possible cause? `warning, fragment not properly served!` doesn't sound good.

> E/GeckoConsole(  514): Content JS LOG at https://marketplace.cdn.mozilla.net/media/js/mkt/consumer-min.js?build=bc854f3-51b10caa:5 in anonymous: warning, fragment not properly served!

Could a networking error get cached?

**Comment by jedp** [comment]
@dougt so if that's the extent of the logs, I'm confused.  The `watch` command is invoked before any sign-in screen is visible.  It registers callbacks.  The `request` command would trigger the sign-in UI to open, and you would see logs referring to `doInternalRequest` etc.

I'm also not sure what to make of the "warning: fragment not properly served" error coming from the marketplace cdn-served content.  Ah.  And @ozten just mentioned that :)

**Comment by vladikoff** [comment]
Got Keon stable build working (`images-keon-2013-06-03.Gecko-08a0e76.Gaia-fcae236`), was able to login to http://firefoxos.123done.org/

OS Version: 1.0.1.0-prerelease
Platform: 18.0
Build: 20130603111001

**Comment by jedp** [comment]
@chilts sent me the logcat from his latest keon build and it looks good.  It seems to have all the fixes we uplifted to v1.0.1 in May.

**Comment by jedp** [comment]
Having spoken with @lloyd, downgrading to four stars.  The workaround is just to upgrade the version of firefoxos on the keon.  But we would like to find a fix that works for everyone if we can.

I think the problem of logging into wheresmyfox should resolve itself when the current persona train is deployed, probably on Monday - see [bug 871799](https://bugzilla.mozilla.org/show_bug.cgi?id=871799).  The reason is that the phone thinks persona is at `firefoxos.persona.org`, not `login.persona.org` (which is where wheresmyfox, 123done, etc. all correctly include the persona include.js from).  Consequently, there's a cross-domain conflict.

Once the current persona train deployment is complete, and we subsequently update our DNS to make `firefoxos.persona.org` into a CNAME for `login.persona.org`, all persona RPs should work properly on the phone.

However, you _should_ be able to log into marketplace successfully on the phone right now.  Marketplace is currently using `firefoxos.persona.org`, so there should be no problems.

Any more logs that people can send to reproduce any of these issues would be appreciated.

Once again, to enable verbose logging, you do this:

``` shell
adb pull /data/local/user.js  # file may not exist.  ok if it doesn't
echo 'pref("toolkit.identity.debug", true);' >> user.js
adb shell stop b2g
adb push pref.js /data/local/user.js
adb shell start b2g   # phone now reboots with new profile
```

**Comment by kumar303** [comment]
The logs here suggest that there is a pretty old Marketplace app on Keon :( It would be nice to see if all Keon users can get on the same 1.0.1-prerelease build as the rest of us.

**Comment by lloyd** [comment]
high level question, is it worth us doing unnatural things to try to fix this, or do we want to rely on update?

Second question, what's the proper forum to message this issue externally?

**Comment by inma610** [comment]
Hi everyone,

@lloyd  I don't know if I can be of much help here, but there is a Mozilla l10n mailing list\* that we localizers use to manage the Persona l10n (among many others), and communicate updates and issues. BTW, a member of the Polish community also reported that he's experiencing the same issue with his Keon.

Perhaps you could contact the rest of Mozilla localizers via that mailing list to help you run different tests. Most of them are Mozilla Reps, so they have a Keon in their hands.

My 2 cents ;)

*https://lists.mozilla.org/listinfo/dev-l10n-web

**Comment by jedp** [comment]
Thanks @inma610.  @lloyd, I'll send to the l10n list the message I sent to mozqa.  Seems like a good start at least for getting the message out to the community.

**Comment by callahad** [comment]
> high level question, is it worth us doing unnatural things to try to fix this, or do we want to rely on update?

I'd vote for not doing unnatural things to support the Keon, especially if a software update solves it. Keons are explicitly labeled as developer preview devices -- I think we're OK asking and expecting those users to upgrade.

If a similar bug hit commercially released phones, on the other hand, I'd be all in favor of us bending over backwards to make it work.

FWIW, I have a Keon now, and will be using it as my sole smartphone, so I'm impacted by whatever decision we make.

**Comment by jedp** [comment]
We've communicated to everyone we can about this and have heard nothing back.  If Keon shipped with an unstable release (pre 1.0.1) of their own build, I don't think there's much we can or should do to try to chase it down.  I'm going to close this as wontfix.  Please reopen and send me hate mail if I'm wrong.
